src/shell/doom.c

```diff
@@ -0,0 +1,231 @@
+        shell_log_errno ("error posting task-exit eventlog entry");
+
+    if (f && doom->timeout != TIMEOUT_NONE)
+        flux_watcher_start (doom->timer);
+
+    flux_future_destroy (f); // fire and forget
+    free (entrystr);
+    json_decref (task_info);
+    flux_kvs_txn_destroy (txn);
+}
+
+static void doom_notify_cb (flux_t *h,
+                            flux_msg_handler_t *mh,
+                            const flux_msg_t *msg,
+                            void *arg)
+{
+    struct shell_doom *doom = arg;
+    json_t *task_info;
+
+    assert (doom->shell->info->shell_rank == 0);
+
+    if (doom->done)
+        return;
+    if (flux_request_unpack (msg, NULL, "o", &task_info) < 0) {
+        shell_log_errno ("error parsing first task exit notification");
+        return;
+    }
+    doom_post (doom, task_info);
+    doom->done = true;
+}
+
+static void doom_notify (struct shell_doom *doom, json_t *task_info)
+{
+    flux_future_t *f;
+
+    assert (doom->shell->info->shell_rank > 0);
+
+    if (!(f = flux_shell_rpc_pack (doom->shell,
+                                   "doom",
+                                   0,
+                                   FLUX_RPC_NORESPONSE,
+                                   "O",
+                                   task_info)))
+        shell_log_errno ("error notifying rank 0 of first task exit");
+    flux_future_destroy (f);
+}
+
+static void doom_timeout (flux_reactor_t *r,
+                          flux_watcher_t *w,
+                          int revents,
+                          void *arg)
+{
+    struct shell_doom *doom = arg;
+    char fsd[64];
+    fsd_format_duration (fsd, sizeof (fsd), doom->timeout);
+    shell_die (1, "%s timeout after first task exit", fsd);
+}
+
+static int doom_task_exit (flux_plugin_t *p,
+                           const char *topic,
+                           flux_plugin_arg_t *args,
+                           void *arg)
+{
+    flux_shell_t *shell;
+    struct shell_doom *doom;
+    flux_shell_task_t *task;
+
+    if (!(shell = flux_plugin_get_shell (p))
+        || !(doom = flux_plugin_aux_get (p, "doom"))
+        || !(task = flux_shell_current_task (shell)))
+        return -1;
+    if (!doom->done) {
+        json_t *task_info;
+
+        if (flux_shell_task_info_unpack (task, "o", &task_info) < 0)
+            return -1;
+        if (shell->info->shell_rank == 0)
+            doom_post (doom, task_info);
+        else
+            doom_notify (doom, task_info);
+        doom->done = true;
+    }
+    return 0;
+}
+
+static void doom_destroy (struct shell_doom *doom)
+{
+    if (doom) {
+        int saved_errno = errno;
+        flux_watcher_destroy (doom->timer);
+        free (doom);
+        errno = saved_errno;
+    }
+}
+
+static int parse_args (flux_shell_t *shell, double *timeoutp)
+{
+    const char *fsd = NULL;
+    double timeout;
+
+    if (flux_shell_getopt_unpack (shell, "exit-timeout", "s", &fsd) < 0)
```

Comment:
Should an integer or real work here as well? (implied seconds), maybe unpack as a json_t object and use json_string_value()?

Reply:
Good idea - that would avoid the somewhat generic error you get now with an integer argument.

Reply:
Oh, just realized it already does since I was using the FSD parser

https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_23.html

I'll push a fix to the value error message though.

Reply:
Oh wait, no it doesn't get past jansson does it.  Fixing.

Reply:
OK, force pushed a change and some more tests.